Create reference-architecture-ccs.md (#816)
terraform-ibm-modules/terraform-ibm-landing-zone: reference-architectures/reference-architecture-ccs.md

---

copyright:
  years: 2024
lastupdated: "2024-06-12"

subcollection: security-services

authors:
- name: Dharmesh Bhakta
  email: bhakta@ibm.com

# The release that the reference architecture describes
version: 1.0

# Use if the reference architecture has deployable code.
# Value is the URL to land the user in the IBM Cloud catalog details page for the deployable architecture.
# See https://test.cloud.ibm.com/docs/get-coding?topic=get-coding-deploy-button

> deployment-url: <url>

Reviewer: As soon as the stack is onboarded to the catalog at cloud.ibm.com/catalog, you can update this URL.

use-case:
  - CloudSecurity
  - KeyManagement
  - CloudSecurityAndCompliance
  - DataCompliance
  - Governance
  - GRC

industry: SoftwareAndPlatformApplications, Technology, Banking, FinancialSector

compliance: CIS Benchmarks

> docs: https://cloud.ibm.com/docs/<subsection>

Reviewer: ```suggestion
docs: https://cloud.ibm.com/docs/security-services
```

content-type: reference-architecture

production: false

---
<!--
The following line inserts all the attribute definitions. Don't delete.
-->
{{site.data.keyword.attribute-definition-list}}

<!--
Don't include "reference architecture" in the following title.
Specify a title based on a use case. If the architecture has a module
or tile in the IBM Cloud catalog, match the title to the catalog. See
https://test.cloud.ibm.com/docs/solution-as-code?topic=solution-as-code-naming-guidance.
-->

> # Core Security Services Stack on IBM Cloud

Reviewer: ```suggestion
# Core security services
```

we don't want to refer to `stack` and this will be viewed on IBM Cloud. 

I assume that it's sentence capitalization because it's not a product.

{: #genai-pattern}
{: toc-content-type="reference-architecture"}

> {: toc-version="1.0"}

Reviewer: BTW. You'll want to update this and the version property above to match the version of the DA (stack) itself when that gets deployed.

<!--
The IDs, such as {: #title-id} are required for publishing this reference architecture in IBM Cloud Docs. Set unique IDs for each heading. Also include
the toc attributes on the H1, repeating the values from the YAML header.
 -->

> This reference architecture summarizes deployment and the best practices for setting of core security sercices and its associated dependency on IBM Cloud. IBM Cloud's core security services are crucial for ensuring robust security and compliance for cloud-based applications and data. Primary goal is to provide framework for secure and compliant IBM Cloud Workloads.

Reviewer: ```suggestion
This reference architecture summarizes the deployment and best practices on IBM Cloud for setting core security services and their associated dependencies. IBM Cloud's core security services are crucial for ensuring robust security and compliance for cloud-based applications and data. Their primary goal is to provide framework for secure and compliant IBM Cloud workloads.
```

Secrets Manager: This service helps in securely storing and managing sensitive information such as API keys, credentials, and certificates. By centralizing secret management, it reduces the risk of exposure and simplifies the process of accessing and rotating secrets, thereby enhancing security posture.

Security and Compliance Center: This platform offers a comprehensive suite of tools to assess, monitor, and maintain the security and compliance of your cloud environment. It provides insights and controls to help organizations meet regulatory requirements, adhere to best practices, and protect against threats.

IBM Cloud Security and Compliance Center Workload Protection: This service offers functionality to protect workloads, get deep cloud and container visibility, posture management (compliance, benchmarks, CIEM), vulnerability scanning, forensics, and threat detection and blocking.

This reference architecture showcases how these services form a foundational security layer that enhances data protection, simplifies compliance, and strengthens overall cloud security for any workload in IBM Cloud. 

## Architecture diagram
{: #architecture-diagram}

> The below diagram represents the architecture for Core Security Services on IBM cloud and reuses the [best practices](https://cloud.ibm.com/docs/framework-financial-services?topic=framework-financial-services-about) for IBM Cloud for Financial Services.
> 
> ![Architecture.](core-security-services-architecture.svg "Architecture"){: caption="Figure 1. Reference Architecture" caption-side="bottom"}

Reviewer: ```suggestion
The following diagram represents the architecture for Core Security Services on IBM Cloud and reuses the [best practices](/docs/framework-financial-services?topic=framework-financial-services-about) for IBM Cloud for Financial Services.

![Architecture.](core-security-services-architecture.svg "Architecture"){: caption="Figure 1. Architecture diagram" caption-side="bottom"}
```

1. Key Protect

Key Protect is responsible for centrally managing the lifecycle of encryption keys used by IBM Cloud Object Storage (COS) buckets, Secrets Manager, and event notification resources. Additionally, it can manage encryption keys for any customer workload requiring protection.

2. Secrets Manager

Secrets Manager securely stores and manages sensitive information, including API keys, credentials, and certificates. It utilizes encryption keys from Key Protect to encrypt sensitive data and to seal/unseal vaults holding the secrets. It is preconfigured to send events to the Event Notifications service, allowing customers to set up email or SMS notifications. Moreover, it is automatically configured to forward all API logs to the customer's logging instance.

3. Security Compliance Center

The Security Compliance Center instance is preconfigured to scan all resources provisioned by the reference architecture. It can be expanded to accommodate the unique workloads of customers.

> Cloud Object Storage (COS) buckets are set up to receive logs from Logging and Alerting Services. Each bucket is configured to encrypt data at rest using encryption keys managed by Key Protect.

Reviewer: ```suggestion
IBM Cloud Object Storage buckets are set up to receive logs from Logging and Alerting Services. Each bucket is configured to encrypt data at rest by using encryption keys managed by Key Protect.
```

Logging and alerting are not Cloud services. Should this be
IBM Cloud Log Analysis and ? ?

|  | [Secrets Manager](https://cloud.ibm.com/docs/secrets-manager?topic=secrets-manager-getting-started#getting-started) | Certificate and Secrets Management |
|  | [Security and Compliance Center (SCC)](https://cloud.ibm.com/docs/security-compliance?topic=security-compliance-getting-started) | Implement controls for secure data and workload deployments, and assess security and compliance posture |
|  | [Security and Compliance Center Workload Protection  ](https://cloud.ibm.com/docs/workload-protection?topic=workload-protection-getting-started) | |
| Service Management | [IBM Cloud Monitoring](https://cloud.ibm.com/docs/monitoring?topic=monitoring-about-monitor) | Apps and operational monitoring |
|  | [IBM Log Analysis](https://cloud.ibm.com/docs/log-analysis?topic=log-analysis-getting-started) | Apps and operational logs |
|  | [Activity Tracker Event Routing](https://cloud.ibm.com/docs/activity-tracker?topic=activity-tracker-getting-started) | Audit logs |
{: caption="Table 2. Components" caption-side="bottom"}

## Compliance
{: #compliance}

> - Ensure Cloud Object Storage encryption is enabled with BYOK
> - Ensure Activity Tracker data is encrypted at rest
> - Ensure Activity Tracker trails are integrated with LogDNA Logs
> - Ensure IBM Key Protect has automated rotation for customer managed keys enabled
> - Ensure the IBM Key Protect service has high availability

Reviewer: ```suggestion
- Ensure Object Storage encryption is enabled with BYOK
- Ensure Activity Tracker data is encrypted at rest
- Ensure Activity Tracker trails are integrated with Log Analysis logs
- Ensure Key Protect has automated rotation enabled for customer-managed keys
- Ensure that the Key Protect service instance is highly available
```

There's no LogDNA any longer, right? assume it's Log Analysis

**Security and Compliance Center (SCC)** <br>

> This reference architecture utilizes the Security and Compliance Center (SCC) which defines policy as code, implements controls for secure data and workload deployments and assess security and compliance posture. For this reference architecture the CIS IBM Cloud Foundations Benchmark is used. A profile is a grouping of controls that can be evaluated for compliance.

Reviewer: ```suggestion
This reference architecture utilizes the Security and Compliance Center, which defines policy as code, implements controls for secure data and workload deployments, and assesses security and compliance posture. For this reference architecture, the CIS IBM Cloud Foundations Benchmark is used. A profile is a grouping of controls that can be evaluated for compliance.
```
